packages/service/lib/stackable.js

```diff
@@ -1,354 +1,243 @@
+      flags: this.configManager.current.plugins?.typescript?.flags,
+      cwd: this.root,
       logger: this.logger
+    })
+  }
+
+  async inject (injectParams, onInject) {
+    const response = await this.#app.inject(injectParams, onInject)
+
+    if (onInject) {
+      return
     }
 
-    await compile(compileOptions)
+    const { statusCode, statusMessage, headers, body } = response
+    return { statusCode, statusMessage, headers, body }
   }
 
-  getUrl () {
-    return this.app !== null ? this.app.url : null
+  getApplication () {
+    return this.#app
   }
 
-  async getInfo () {
-    const type = this.stackable.configType
-    const version = this.stackable.configManagerConfig.version ?? null
-    return { type, version }
+  async getDispatchFunc () {
+    await this.init()
+    return this.#app
   }
 
   async getConfig () {
-    const config = Object.assign({}, this.configManager.current)
-    config.server = Object.assign({}, config.server)
-
-    const logger = config.server.loggerInstance
+    const loggerInstance = this.serverConfig?.loggerInstance
 
-    if (logger) {
-      config.server.logger = {}
+    if (loggerInstance) {
+      const config = Object.assign({}, this.configManager.current)
+      const { loggerInstance: _, ...serverConfig } = this.serverConfig
+      config.server = { ...serverConfig, logger: { level: loggerInstance.level } }
 
-      if (logger.level) {
-        config.server.logger.level = logger.level
-      }
+      return config
     }
 
-    delete config.server.loggerInstance
-
-    return config
-  }
-
-  async getEnv () {
-    return this.configManager.env
-  }
-
-  getMeta () {
-    const config = this.configManager.current
-
-    return {
-      composer: {
-        prefix: config.basePath ?? this.basePath ?? this.context?.serviceId,
-        wantsAbsoluteUrls: false,
-        needsRootTrailingSlash: false,
-        tcp: !!this.app?.url,
-        url: this.app?.url
-      },
-      connectionStrings: [this.connectionString]
-    }
+    return super.getConfig()
   }
 
   async getWatchConfig () {
     const config = this.configManager.current
 
     const enabled = config.watch?.enabled !== false && config.plugins !== undefined
 
+    if (!enabled) {
+      return { enabled, path: this.root }
+    }
+
     return {
       enabled,
-      path: this.configManager.dirname ?? dirname(this.configManager.fullPath),
+      path: this.root,
       allow: config.watch?.allow,
       ignore: config.watch?.ignore
     }
   }
 
-  async getDispatchFunc () {
-    await this.init()
-    return this.app
-  }
-
-  async getDispatchTarget () {
-    return this.getUrl() ?? (await this.getDispatchFunc())
+  getMeta () {
+    return {
+      composer: {
+        tcp: typeof this.url !== 'undefined',
+        url: this.url,
+        prefix: this.basePath ?? this.#basePath,
+        wantsAbsoluteUrls: false,
+        needsRootTrailingSlash: false
+      },
+      connectionStrings: [this.connectionString]
```

Comment:
What is connectionString?

Reply:
It was set in https://github.com/platformatic/platformatic/pull/3577.